server/cmd/code-annotation/handler/assignments.go

```diff
@@ -0,0 +1,74 @@
+package handler
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/render"
+
+	"github.com/src-d/code-annotation/server/cmd/code-annotation/serializer"
+	"github.com/src-d/code-annotation/server/model"
+	"github.com/src-d/code-annotation/server/repository"
+)
+
+func HandleGetAssignmentsForUserExperiment(w http.ResponseWriter, r *http.Request) {
+	experimentID, err := strconv.Atoi(chi.URLParam(r, "experimentId"))
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	userID, err := strconv.Atoi("1")
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	assignments, err := repository.GetAssignmentsFor(userID, experimentID)
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	render.Render(w, r, &serializer.ResponseMany{Data: serializer.NewAssignments(assignments)})
+}
+
+func HandleSaveAssignment(w http.ResponseWriter, r *http.Request) {
+	assignmentId, err := strconv.Atoi(chi.URLParam(r, "assignmentId"))
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+	if err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	var assignment serializer.Assignment
+	if err := json.Unmarshal(body, &assignment); err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	answer := model.Answer(assignment.Answer)
+	if err := repository.SaveAssignment(assignmentId, answer, assignment.Duration); err != nil {
+		render.Render(w, r, serializer.NewErrors(err))
+		return
+	}
+
+	//TODO: use chi logger
```

Comment:
can we use logrus?

Reply:
I'd use chi default through `chi/middleware/Logger`, whose default seems to be `STDOUT` and `STDERR`, but also lets us use Logrus or others.

I'd prefer to consider the Logging lib to use in a separate issue since we're using three big logging libraries in different of our biggest projects
- `log` &rarr; go-git...
- `github.com/inconshreveable/log15` &rarr; borges, rovers, framework...
- `github.com/sirupsen/logrus` &rarr; go-compose-installer

Reply:
`chi/middleware/Logger` the name of the package says you it's not supposed to be used outside of middleware. There is no problem to use middleware with logrus: https://github.com/go-chi/chi/blob/master/_examples/logging/main.go

`log15` is also fine for me, can we vote`logrus` vs`log15` then?
for go-git using standard log makes sense. it's library, not application.

Reply:
I didn't test it yet, but [if I understood it](https://github.com/go-chi/chi/blob/master/_examples/logging/main.go#L132) the Logger can be retrieved (and used) from the inside of the handlers;
```go
type printfer interface{
  func Printf(format string, v ...interface{})
}
```
and then
```go
logger := r.Context().Value(middleware.LogEntryCtxKey).(printfer)
logger.Printf("whatever")
```

About the log library to use, I think we should consider it as a team decision, considering the company first rule; I mean, imho we should use the same logger in all our projects, and it should be the one that is "the most wanted" by engineering. What do you think about that decision @bzz ?

Reply:
Let's use Logrus for consistency.

Reply:
That's what I was trying to find: [Choose one log library for Go and implement on all the projects](https://github.com/src-d/backlog/issues/840#issuecomment-343930191), many thanks for pointing it out !!